### ISSUE TITLE
restic check -> error: hex.DecodeString: encoding/hex: odd length hex string

### ISSUE THREAD

**Issue by holli** [restic/restic#2060]
Output of `restic version`
--------------------------
restic 0.9.0 compiled with go1.10.2 on linux/amd64

I also tried "restic check" with latest pull from github: "restic 0.9.3-dev (compiled manually) compiled with go1.11.1 on linux/amd64" . No differences.

How did you run restic exactly?
-------------------------------
export RESTIC_REPOSITORY="sftp: ...
export RESTIC_PASSWORD=" ...

restic check

```
using temporary cache in /tmp/restic-check-cache-660986461
repository c20a0ff7 opened successfully, password is correct
create exclusive lock for repository
load indexes
check all packs
check snapshots, trees and blobs
error: hex.DecodeString: encoding/hex: odd length hex string
Fatal: repository contains errors
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------

sftp connected to linux server 

Expected behavior
-----------------

Not have fatal errors :P  At least have some information what is wrong and how to fix it.

Actual behavior
---------------

restic check outputs: "error: hex.DecodeString: encoding/hex: odd length hex string"

I did run "restic prune && restic rebuild-index" but they did not change the result.

Do you have any idea what may have caused this?
-----------------------------------------------
I don't know what exactly. I installed restic only couple days ago, backuped about 200gb of different folders and everything seemed ok for few days. I had some connection issues to the server sometimes when playing around.

I can't see anything informative in the debug log when I run with custom build. The debug file doesn't even include the original error I see when running the command. Only lines that contain "error" or "fatal" or "hex" are:

- 2018/10/23 14:46:29 repository/key.go:136       repository.SearchKey.func1      1       key 4317...  returned error ciphertext verification failed
- ...
- 2018/10/23 14:46:51 checker/checker.go:111      checker.(*Checker).LoadIndex.func2      14      loading indexes finished, error: <nil>

**Comment by holli** [comment]
Btw I noticed this error when I tried to tag snapshot. Currently it results in the following error

```
restic tag --add keep 7846d9c6

repository c20a0ff7 opened successfully, password is correct
create exclusive lock for repository
panic: runtime error: slice bounds out of range

goroutine 66 [running]:
github.com/restic/restic/internal/restic.Find.func1(0x0, 0xc4202e035d, 0x3, 0x3, 0xc4202ee8f8)
	src/github.com/restic/restic/internal/restic/backend_find.go:27 +0xcf
github.com/restic/restic/internal/backend.(*RetryBackend).List.func1.1(0x0, 0xc4202e035d, 0x3, 0x3, 0x1)
	src/github.com/restic/restic/internal/backend/backend_retry.go:133 +0xd7
github.com/restic/restic/internal/backend/sftp.(*SFTP).List(0xc420114090, 0xc06fa0, 0xc4202766c0, 0xb79dc0, 0x8, 0xc42025a3f0, 0x70a101, 0xc42025a3f0)
	src/github.com/restic/restic/internal/backend/sftp/sftp.go:460 +0x313
github.com/restic/restic/internal/backend.(*RetryBackend).List.func1(0xc420164c20, 0xc420164c20)
	src/github.com/restic/restic/internal/backend/backend_retry.go:127 +0x12a
github.com/restic/restic/vendor/github.com/cenkalti/backoff.RetryNotify(0xc42006b9d8, 0xc01860, 0xc420164c20, 0xc42006b920, 0xc420164c20, 0xc420164c20)
	src/github.com/restic/restic/vendor/github.com/cenkalti/backoff/retry.go:37 +0x88
github.com/restic/restic/internal/backend.(*RetryBackend).retry(0xc420270720, 0xc06fa0, 0xc420276700, 0xc42233ecb0, 0xe, 0xc4200739d8, 0xe, 0xc420073a38)
	src/github.com/restic/restic/internal/backend/backend_retry.go:36 +0x138
github.com/restic/restic/internal/backend.(*RetryBackend).List(0xc420270720, 0xc06fa0, 0xc4202766c0, 0xb79dc0, 0x8, 0xc420164bc0, 0x0, 0x0)
	src/github.com/restic/restic/internal/backend/backend_retry.go:126 +0x235
github.com/restic/restic/internal/restic.Find(0x7ff76a56b9c0, 0xc42014ea80, 0xb79dc0, 0x8, 0x7ffc6c495180, 0x8, 0x0, 0x0, 0x0, 0x0)
	src/github.com/restic/restic/internal/restic/backend_find.go:26 +0x136
github.com/restic/restic/internal/restic.FindSnapshot(0xc0f200, 0xc4202bb0a0, 0x7ffc6c495180, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	src/github.com/restic/restic/internal/restic/snapshot_find.go:75 +0xb3
main.FindFilteredSnapshots.func1(0xc4202f8600, 0xc42026cae0, 0x1, 0x3, 0xc06fa0, 0xc420276680, 0xc4202bb0a0, 0x0, 0x0, 0x0, ...)
	src/github.com/restic/restic/cmd/restic/find.go:32 +0x1af
created by main.FindFilteredSnapshots
	src/github.com/restic/restic/cmd/restic/find.go:13 +0x126
```

**Comment by fd0** [comment]
Uh, that's a bug, restic should not panic in both cases. I suspect there are files not created by restic in the repository. Can you try running the following command on the directory which contains the repo on the server:
```
$ find /path/to/repo -type f | egrep -v '/[a-f0-9]{64}'
```

This should give us a hint how to reproduce the issue, so we can fix it.

**Comment by holli** [comment]
Amazingly fast debugging!  :)

```
find . -type f | egrep -v '/[a-f0-9]{64}'
./config
./snapshots/100
```
snapshots/100 was empty file with different file permissions than rest of the files. After removing it away everything seemed to work ok. So it might be result of my own script creating an empty file at some point (I made some scripts to check quota around the timestamp of that file).  Couple pointers anyway: `check ` could also check that this is in order or fail with hinting stacktrace; or the code that parses snapshots/hex could ignore / log better the error.

Seems to be resolved for me \o/

**Comment by MichaelEischer** [comment]
Closing this as the bug seems to have been fixed in restic 0.9.5. Since then I'm no longer able to reproduce this issue. Feel free to complain if this is still a problem.
